scripts/crowbar-testbuild.py

```diff
@@ -100,59 +109,109 @@ def add_pr_to_checkout(repo, pr_id, head_sha1, pr_branch, spec):
         repo, pr_id, head_sha1))
 
 
-def trigger_testbuild(repo, github_opts):
-    pr_id, head_sha1, pr_branch = github_opts.split(':')
+def prep_osc_dir(workdir, org, repo, pr_id, head_sha1, pr_branch, pkg, spec):
+    os.chdir(workdir)
+    iosc('co', IBS_MAPPING[pr_branch], pkg, '-c')
+    os.chdir(pkg)
+    add_pr_to_checkout(org, repo, pr_id, head_sha1, pr_branch, spec)
+
+
+def prep_webroot(ptfdir):
+    webroot = os.path.join(htdocs_dir, ptfdir)
+    shutil.rmtree('-rf', webroot)
+    if not os.path.isdir(webroot):
+        os.makedirs(webroot)
+    return webroot
+
+
+def build_package(spec, webroot, pr_branch):
+    buildroot = os.path.join(os.getcwd(), 'BUILD')
+    repository = 'SLE_12' if pr_branch == 'master' else 'SLE_11_SP3'
+
+    dst_log = os.path.join(webroot, 'build.log')
+    out = open(dst_log, 'w')
+
+    # Write output to webroot as we go, so that we can follow
+    # progress directly from the link in the github PR web UI.
+    def tee(line):
+        print line,
+        out.write(line)
+        out.flush()
+
+    try:
+        iosc('build',
+             '--root', buildroot,
+             '--noverify',
+             '--noservice',
+             repository, 'x86_64', spec,
+             _out=tee)
+    finally:
+        out.close()
+
+    sh.cp('-p',
+          sh.glob(os.path.join(buildroot,
+                               '.build.packages/RPMS/*/*.rpm')),
+          webroot)
+
 
-    olddir = os.getcwd()
+def trigger_testbuild(args):
     workdir = tempfile.mkdtemp()
     build_failed = False
+    pkg = args.repo if args.repo == "crowbar" else "crowbar-" + args.repo
+    spec = pkg + '.spec'
+    ptfdir = ':'.join([args.repo, args.pr_id, args.sha1, args.branch])
+    ptfdir, n = re.subn('/', '_', ptfdir)
```

Comment:
ptfdir = ptfdir.replace('/', '_') (btw the slashes in there totally work fine..)